library/Jira/Web/RenderingHelper.php

```diff
@@ -3,63 +3,179 @@
+    }
 
-        return new Link([new Icon('laptop'), $host], Url::fromPath('monitoring/host/show', [
-            'host' => $host
-        ]), [
-            'title' => t('Show Icinga Host State'),
-        ]);
+    /**
+     * Get the link of monitored host
+     *
+     * @return ?Link
+     */
+    public function getHostLink()
+    {
+        return $this->hostLink;
     }
 
-    public function linkToMonitoringService($host, $service)
+    /**
+     * Get the link of monitored service
+     *
+     * @return ?Link
+     */
+    public function getServiceLink()
     {
-        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
-            return new Link([new Icon('cog'), $service], Url::fromPath('icingadb/service', [
-                'name'          => $service,
-                'host.name'     => $host,
-            ]), [
-                'title' => t('Show Icinga Service State'),
-            ]);
-        }
+        return $this->serviceLink;
+    }
 
-        return new Link([new Icon('cog'), $service], Url::fromPath('monitoring/service/show', [
-            'host'    => $host,
-            'service' => $service,
-        ]), [
-            'title' => t('Show Icinga Service State'),
-        ]);
+
+    /**
+     * Get the formatted issue comment using author, time and description or comment body
+     *
+     * @param string|HtmlElement[] $author
+     * @param string $time
+     * @param string $body
+     *
+     * @return HtmlElement[]
+     */
+    public function getIssueComment($author, string $time, string $body)
+    {
+        return [
+            Html::tag('h3', null, Html::sprintf(
+                '%s: %s',
+                $this->shortTimeSince($time),
+                $author
+            )),
+            Html::tag(
+                'pre',
+                ['class' => 'comment'],
+                $this->formatBody($body)
+            ),
+        ];
+    }
+
+    /**
+     * Format the given the issue description or comment body
+     *
+     * @param string $body
+     *
+     * @return HtmlString
+     */
+    public function formatBody(string $body)
+    {
+        $html = Html::wantHtml($body)->render();
+
+        // This is safe.
+        return new HtmlString($this->replaceLinks($html) ?? '');
+    }
+
+    /**
+     * Replace object urls in the given string with link elements
+     *
+     * @param string $string
+     *
+     * @return ?string
+     */
+    protected function replaceLinks($string)
+    {
+        return preg_replace_callback('/\[([^|]+)\|([^]]+)]/', function ($match) {
+            $url = Url::fromPath(htmlspecialchars_decode($match[2]));
+            $link = (new Link(
+                $match[1],
+                $url,
+                ['target' => '_blank']
+            ));
+
+            if ($url->hasParam('service') || $url->hasParam('host.name')) {
+                if (str_contains($match[2], 'monitoring') && $this->useIcingaDbAsBackend()) {
```

Comment:
I have no idea why none of the tooling didn't complain about this, but `str_contains` is php 8 only.